dbms/src/Storages/DeltaMerge/File/DMFilePackFilter.cpp

```diff
@@ -108,14 +98,47 @@ void DMFilePackFilter::init()
     LOG_DEBUG(
         log,
         "RSFilter exclude rate: {:.2f}, after_pk: {}, after_read_packs: {}, after_filter: {}, handle_ranges: {}"
-        ", read_packs: {}, pack_count: {}",
+        ", read_packs: {}, pack_count: {}, none_count: {}, some_count: {}, all_count: {}",
         ((after_read_packs == 0) ? std::numeric_limits<double>::quiet_NaN() : filter_rate),
         after_pk,
         after_read_packs,
         after_filter,
         toDebugString(rowkey_ranges),
         ((read_packs == nullptr) ? 0 : read_packs->size()),
-        pack_count);
+        pack_count,
+        none_count,
+        some_count,
+        all_count);
+}
+
+std::tuple<UInt64, UInt64, UInt64> DMFilePackFilter::countPackRes() const
+{
+    UInt64 none_count = 0;
+    UInt64 some_count = 0;
+    UInt64 all_count = 0;
+    for (auto res : pack_res)
+    {
+        switch (res)
+        {
+        case RSResult::None:
+            ++none_count;
+            break;
+        case RSResult::Some:
+            ++some_count;
+            break;
+        case RSResult::All:
+            ++all_count;
+            break;
+        default:
+            throw Exception(ErrorCodes::LOGICAL_ERROR, "{} is invalid", magic_enum::enum_name(res));
```

Comment:
```suggestion
            throw Exception(ErrorCodes::LOGICAL_ERROR, "{} is invalid", res);
```

magic_enum::enum_name always return `` here.